webextensions/sidebar/components/TSTCloseBoxElement.js

```diff
@@ -0,0 +1,47 @@
+const NORMAL_TOOLTIP = 'tab_closebox_tab_tooltip';
+const MULTISELECTED_TOOLTIP = 'tab_closebox_tab_tooltip_multiselected';
+const TREE_TOOLTIP = 'tab_closebox_tree_tooltip';
+
+export const CloseBoxTooltipType = Object.freeze({
+  Normal: 'normal',
+  MultiSelected: 'multiselected',
+  Tree: 'tree',
+});
+
+function getTooltipLabelKey(tooltipType) {
+  switch (tooltipType) {
+    case CloseBoxTooltipType.Normal:
+      return NORMAL_TOOLTIP;
+    case CloseBoxTooltipType.MultiSelected:
+      return MULTISELECTED_TOOLTIP;
+    case CloseBoxTooltipType.Tree:
+      return TREE_TOOLTIP;
+    default:
+      throw new RangeError(`${tooltipType} is not unknown TooltipType`);
+  }
+}
+
+export class TSTCloseBoxElement extends HTMLElement {
+  constructor() {
+    super();
+
+    this._isInitialized = false;
+  }
+
+  connectedCallback() {
+    if (this._isInitialized) {
+      return;
+    }
+
+    this.setAttribute('title', browser.i18n.getMessage(NORMAL_TOOLTIP));
+    this.setAttribute('draggable', true); // this is required to cancel click by dragging
```

Comment:
* If we do these operations in a `constructor` of this class, Gecko throws `NotSupportedError: Operation is not supported`. 
* `connectedCallback()` may be called multiple times [by append/remove operations](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements).
* `browser.i18n.getMessage()`might be a costly operation.

So I make ensure to call these operation only once conservatively.

Reply:
Thank you for explanation! This information (especially the first one) looks good to be written as a comment with the URL of the corresponding bug on the BMO.

Reply:
I'm not familiar with details of the spec, but this is not Gecko's bug.

> 6. If result has children, then throw a "NotSupportedError" DOMException. 
>
> https://dom.spec.whatwg.org/#concept-create-element

Reply:
Thanks, I misunderstood that the "NotSupported" means "Gecko does not support it". I got that it actually means "such an operation is not supported in the spec." Then I think that a comment with the URL of the spec may help future work for me until I master the WebComponents spec completely.

Reply:
I've added the comment: https://github.com/piroor/treestyletab/pull/2392/commits/3402277a619631f7f258afa430a0d0ad084235c0